### PR TITLE
Try to fix #6579

### LINKED ISSUE #6579: afta segfault
``` 
perl -e 'print "afta \x02\90\x80\x42"x4008' | r2 -n - 
 -- You haxor! Me jane?
Segmentation fault (core dumped)
,gdb -q r2 core 
Core was generated by `r2 -n -'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fe1c733ec51 in r_line_readline_cb (cb=0x0, user=0x0) at dietline.c:1018
1018                            while (s > I.buffer.data && (*s & 0xc0) == 0x80) {
(gdb) bt
#0  0x00007fe1c733ec51 in r_line_readline_cb (cb=0x0, user=0x0) at dietline.c:1018
#1  0x00007fe1c733e8e5 in r_line_readline () at dietline.c:491
#2  0x00007fe1c79f0b89 in r_core_fgets (buf=0x7ffc4a7ff390 "", len=4096) at core.c:1139
#3  0x00007fe1c7341ea8 in r_cons_fgets (buf=0x7ffc4a7ff390 "", len=4096, argc=0, argv=0x0) at input.c:224
#4  0x00007fe1c79f3640 in r_core_prompt (r=0x55e68e1a95a0 <r>, sync=0) at core.c:1849
#5  0x000055e68dfa6395 in main (argc=3, argv=0x7ffc4a800aa8, envp=0x7ffc4a800ac8) at radare2.c:1124
(gdb) l
1013                    case 2: // ^b // emacs left
1014    #if USE_UTF8
1015                    {
1016                            char *s = I.buffer.data+I.buffer.index-1;
1017                            utflen = 1;
1018                            while (s > I.buffer.data && (*s & 0xc0) == 0x80) {
1019                                    utflen++;
1020                                    s--;
1021                            }
1022                     }
(gdb) x/20i $pc
=> 0x7fe1c733ec51 <r_line_readline_cb+874>:     movzbl (%rax),%eax
   0x7fe1c733ec54 <r_line_readline_cb+877>:     movsbl %al,%eax
   0x7fe1c733ec57 <r_line_readline_cb+880>:     and    $0xc0,%eax
   0x7fe1c733ec5c <r_line_readline_cb+885>:     cmp    $0x80,%eax
   0x7fe1c733ec61 <r_line_readline_cb+890>:     je     0x7fe1c733ec34 <r_line_readline_cb+845>
   0x7fe1c733ec63 <r_line_readline_cb+892>:     mov    0x20f7a7(%rip),%eax        # 0x7fe1c754e410 <r_line_instance+4144>
   0x7fe1c733ec69 <r_line_readline_cb+898>:     test   %eax,%eax
   0x7fe1c733ec6b <r_line_readline_cb+900>:     je     0x7fe1c733ec7b <r_line_readline_cb+916>
   0x7fe1c733ec6d <r_line_readline_cb+902>:     mov    0x20f79d(%rip),%eax        # 0x7fe1c754e410 <r_line_instance+4144>
   0x7fe1c733ec73 <r_line_readline_cb+908>:     sub    -0x94(%rbp),%eax
   0x7fe1c733ec79 <r_line_readline_cb+914>:     jmp    0x7fe1c733ec80 <r_line_readline_cb+921>
   0x7fe1c733ec7b <r_line_readline_cb+916>:     mov    $0x0,%eax
   0x7fe1c733ec80 <r_line_readline_cb+921>:     mov    %eax,0x20f78a(%rip)        # 0x7fe1c754e410 <r_line_instance+4144>
   0x7fe1c733ec86 <r_line_readline_cb+927>:     jmpq   0x7fe1c733fc0c <r_line_readline_cb+4901>
   0x7fe1c733ec8b <r_line_readline_cb+932>:     cmpb   $0x18,-0x99(%rbp)
   0x7fe1c733ec92 <r_line_readline_cb+939>:     jne    0x7fe1c733ed50 <r_line_readline_cb+1129>
   0x7fe1c733ec98 <r_line_readline_cb+945>:     mov    0x20f776(%rip),%eax        # 0x7fe1c754e414 <r_line_instance+4148>
   0x7fe1c733ec9e <r_line_readline_cb+951>:     movslq %eax,%rdx
   0x7fe1c733eca1 <r_line_readline_cb+954>:     lea    0x20e768(%rip),%rax        # 0x7fe1c754d410 <r_line_instance+48>
   0x7fe1c733eca8 <r_line_readline_cb+961>:     movb   $0x0,(%rdx,%rax,1)
(gdb) x/10x $pc
0x7fe1c733ec51 <r_line_readline_cb+874>:        0x0f00b60f      0xc025c0be      0x3d000000      0x00000080
0x7fe1c733ec61 <r_line_readline_cb+890>:        0x058bd174      0x0020f7a7      0x0e74c085      0xf79d058b
0x7fe1c733ec71 <r_line_readline_cb+906>:        0x852b0020      0xffffff6c
```

### PR DESCRIPTION
**Pull request by leberus**
This commit tries to address #6579
I've checked it out on my side, and I doesn't segfault anymore. I couldn't reproduce the reports either.
I've also done some checks and it doesn't break anything.

### CODE REVIEW

File: libr/cons/dietline.c [hunk 1]
```diff
@@ -1465,9 +1465,14 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 #endif
 			}
 #if USE_UTF8
-			I.buffer.index += utflen;
+			if ((I.buffer.index + utflen) <= I.buffer.length &&
+				(I.buffer.index + utflen) >= utflen) {
```
Comment (radare) [inline libr/cons/dietline.c]: i dont get the logic behind x + y >= x

Comment (leberus) [inline libr/cons/dietline.c]: my bad, was something i added while testing, i'm gonna take it out now

File: libr/cons/dietline.c [hunk 2]
```diff
@@ -1465,9 +1465,13 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 #endif
 			}
 #if USE_UTF8
-			I.buffer.index += utflen;
+			if ((I.buffer.index + utflen) <= I.buffer.length) {
+					I.buffer.index += utflen;
+			}
 #else
-			I.buffer.index++;
+			if ((I.buffer.index + 1) <= I.buffer.length) {
```
Comment (radare) [inline libr/cons/dietline.c]: we can simplify this expression with: `if (I.buffer.index < I.buffer.length)`

**Comment by leberus** [comment]
Simplified

**Pull request merged**